app/src/org/commcare/activities/CommCareWiFiDirectActivity.java

```diff
@@ -659,9 +730,9 @@ protected void deliverError(CommCareWiFiDirectActivity receiver, Exception e) {
             }
 
             @Override
-            protected void deliverResult(CommCareWiFiDirectActivity receiver, FormRecord[] result) {
-                if (result != null) {
-                    receiver.onZipSuccesful(result);
+            protected void deliverResult(CommCareWiFiDirectActivity receiver, Integer result) {
+                if (result == ZipTask.RESULT_SUCCESS) {

```

Comment:
Do you need to unbox the result with an `intValue()`? I'm not sure, but you manually unbox in other places.